crates/ink/codegen/src/generator/metadata.rs

```diff
@@ -256,7 +264,7 @@ impl Metadata<'_> {
                 let mutates = message.receiver().is_ref_mut();
                 let ident = message.ident();
                 let args = message.inputs().map(Self::generate_dispatch_argument);
-                let ret_ty = Self::generate_return_type(message.output());
+                let ret_ty = Self::generate_return_type(Some(&message.wrapped_output()));
```

Comment:
What if the metadata crate would transparently "wrap" the return type spec into a `Result<message::Output, ::ink::LangError> for you? On a first glance this _should_ be possible in `ContractSpecBuilder::done()`, since the language error is available in the contract spec.

It would make it a slightly more pleasant API (especially for other languages where there is no `Result` type available) and it would ensure that all message return type specs are wrapped correctly in the metadata.

Reply:
So I think the `ReturnTypeSpec` from the metadata crates deserves a refactor. I started with it here. However I'd wait for #1491 first.

Also since this is just some metadata API change I think it should not hold up our beta release.